consensus/src/chained_bft/event_processor.rs

```diff
@@ -513,95 +512,40 @@ impl<T: Payload, P: ProposerInfo> EventProcessor<T, P> {
         {
             counters::CREATION_TO_RECEIVAL_MS.observe(time_to_receival.as_millis() as f64);
         }
-        let block = match self
-            .sync_manager
-            .execute_and_insert_block(proposal.proposal)
```

Comment:
I remember we moved the execution before checking pacemaker before to avoid rejecting valid blocks and requesting from peers if pacemaker is moved by new QC

Reply:
Right, thanks! Let me move it back. I'd like to also change the current logic so that we'd have some dedicated NIL authors instead of using the genesis signers.